nltk/model/counter.py

```diff
@@ -0,0 +1,99 @@
+# Natural Language Toolkit: Language Model Counters
+#
+# Copyright (C) 2001-2016 NLTK Project
+# Author: Ilia Kurenkov <ilia.kurenkov@gmail.com>
+# URL: <http://nltk.org/>
+# For license information, see LICENSE.TXT
+
+from __future__ import unicode_literals
+
+from collections import Counter, defaultdict
+from itertools import chain
+
+from nltk.util import ngrams
+from nltk.probability import FreqDist, ConditionalFreqDist
+from nltk import compat
+
+
+@compat.python_2_unicode_compatible
+class LanguageModelVocabulary(Counter):
+    """Stores language model vocabulary.
+
+    Satisfies two common language modeling requirements for a vocabulary:
+    - When checking membership and calculating its size, filters items by comparing
+      their counts to a cutoff value.
+    - Adds 1 to its size so as to account for "unknown" tokens.
+    """
+
+    def __init__(self, unknown_cutoff, *counter_args):
+        Counter.__init__(self, *counter_args)
+        self.cutoff = unknown_cutoff
+
+    def __contains__(self, item):
+        """Only consider items with counts GE to cutoff as being in the vocabulary."""
+        return self[item] >= self.cutoff
```

Comment:
No worries! Revisiting that actually made me aware of a potential subtle bug: if someone (for whatever crazy reason) sets the cutoff to `0`, the vocabulary will silently start returning `True` even for words unseen during training.
I wonder if the best way to mitigate this is by making sure the value of `cutoff` is at least equal to 1 and raising some clear exception if that's not the case so as to stop the user in their tracks before things get weird.